### PR TITLE
[Quest] The Fanged One Interaction Framework

### PR DESCRIPTION
**Pull request by hooksta4**
[Quest] The Fanged One Interaction Framework

**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
Converts the quest The Fanged One (RNG Flag) to Interaction Framework
- Adds functionality for Old Sabertooth to path in the hallway.
- Adjusted the poison effect to ~2%/tic as per the capture.
- Added listener to Old Sabertooth (in the mob lua file) to capture if players damaged the mob. If they do, they incur a 5 min wait timer to respawn. 
- Replaced NPC lua events into DefaultActions.lua
- Capture shows black tiger fang being traded and taken during quest.

## Steps to test these changes
Be on any job 30+ and speak with Perih Vasha.

Capture located at 
https://www.youtube.com/watch?v=FWtjEDaDG7s

### CODE REVIEW

**Comment by Xaver-DaRed** [comment]
Arent you removing all the logic for several other quests?

**Comment by hooksta4** [comment]
> Arent you removing all the logic for several other quests?

Well shit -  yes I am I was going to PR the entire quest series but opted to just do that one to make it an easier PR to read.

File: scripts/quests/windurst/The_Fanged_One.lua [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+        [xi.zone.WINDURST_WOODS] =
+        {
+            ['Perih_Vashai'] = quest:progressEvent(351),
+
+            onEventFinish =
+            {
+                [351] = function(player, csid, option, npc)
+                    quest:begin(player)
+                    quest:setVar(player, 'Prog', 1)
```
Comment (Xaver-DaRed) [inline scripts/quests/windurst/The_Fanged_One.lua]: We dont need to set prog to 1 if the quest has just started.

File: scripts/zones/Sauromugue_Champaign/npcs/Tiger_Bones.lua [hunk 2]
```diff
@@ -12,30 +12,6 @@ entity.onTrade = function(player, npc, trade)
...
-        fangedOneCS == 2 and
-        not player:hasKeyItem(xi.ki.OLD_TIGERS_FANG)
-    then
-        player:addKeyItem(xi.ki.OLD_TIGERS_FANG)
-        player:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.OLD_TIGERS_FANG)
-        player:setCharVar('TheFangedOneCS', 0)
-
-    -- DEFAULT DIALOG
-    else
-        player:messageSpecial(ID.text.NOTHING_HAPPENS)
```
Comment (Xaver-DaRed) [inline scripts/zones/Sauromugue_Champaign/npcs/Tiger_Bones.lua:37]: Shouldnt this be added to this zone DefaultActions?

And I mean, the actual default outside the quest (before and after). I can go check it real quick if you need it

**Comment by Xaver-DaRed** [comment]
Ok, some info:
- Default Action for Bones in Sauromuge -> Message 7242
imagen
- Default action for Muhk Johldy AT LEVEL 11 and Windurst rank 1, Bastok rank 6 -> 339
imagen
- Default action for Kapeh Myohrye AT LEVEL 11 and Windurst rank 1, Bastok rank 6 -> 340 with params and an event update!
imagen
- Default action for Perih Vashai AT LEVEL 11 and Windurst rank 1, Bastok rank 6 -> 338
imagen

So from what I am seing, level is not a defining factor in the defaults, but SOMETHING is

Same events for guards pre-quest at proper level. So I am guessing it has to do with Sybil storyline or something

File: scripts/zones/Sauromugue_Champaign/npcs/Tiger_Bones.lua (hunk 2 above)
Comment (hooksta4) [reply scripts/zones/Sauromugue_Champaign/npcs/Tiger_Bones.lua:37]: The capture didnt show that message, it always said There seems to be bones here.

**Comment by hooksta4** [comment]
> Ok, some info:
> 
> * Default Action for Bones in Sauromuge -> Message 7242
>   imagen
> * Default action for Muhk Johldy AT LEVEL 11 and Windurst rank 1, Bastok rank 6 -> 339
>   imagen
> * Default action for Kapeh Myohrye AT LEVEL 11 and Windurst rank 1, Bastok rank 6 -> 340 with params and an event update!
>   imagen
> * Default action for Perih Vashai AT LEVEL 11 and Windurst rank 1, Bastok rank 6 -> 338
>   imagen
> 
> So from what I am seing, level is not a defining factor in the defaults, but SOMETHING is
> 
> Same events for guards pre-quest at proper level. So I am guessing it has to do with Sybil storyline or something

Ok I mean that makes sense. I would imagine that when those specifics get reworked into interaction framework they will be addressed? Additionally, if there is a condition, then the DefaultAction would need to be removed and placed into their specific lua files yes?

**Comment by Xaver-DaRed** [comment]
> Ok I mean that makes sense. I would imagine that when those specifics get reworked into interaction framework they will be addressed? Additionally, if there is a condition, then the DefaultAction would need to be removed and placed into their specific lua files yes?

DefaultActions are meant to be defaults. As in, "What message would a brand new character get?"

The messages you placed in there, while can happen, arent defaults. My guess is they are nation rank based, but its just a guess.

I'm gonna create a brand new character just for this, however, we can go 2 ways with this:
- First option:
  - In the default action file, we place the actual defaults.
  - In the quest script, in the "quest available" section, we place the rank check and the diferent dialogues for the guards.
  - In the quest script, in the "quest completed" section, we use do the same, using :replaceDefault as needed.

If we do it this way, we dont need any logic in the npc scripts.

- Second option:
  - ALL logic, including this quest logic for this npcs would go to their npc lua scripts

I dont like the second one, it goes against what the interaction framework is for: not having quest logic segregated.

**Comment by hooksta4** [comment]
Totally get that. I was going based off of captures. I just went on and got this 
image


This char is rank 6 in Windy.

**Comment by hooksta4** [comment]
Someone from bastok provided me with 

image

image

Bastok citizen.

**Comment by Xaver-DaRed** [comment]
Brand new Windurst character
img_20240411_025023

**Comment by hooksta4** [comment]
Ok made those changes to default however, I think that it might be out of scope for this PR to figure out which mission/quest changes that dialog. It does appear to have to do something with windurst nation as my  cap showed the other "default".

**Comment by hooksta4** [comment]
Also, did you see me standing there?

File: scripts/zones/Sauromugue_Champaign/npcs/Tiger_Bones.lua (hunk 2 above)
Comment (Xaver-DaRed) [reply scripts/zones/Sauromugue_Champaign/npcs/Tiger_Bones.lua:37]: I gave the proper message below. Its the bones one.

It does need to go to the default actions.

**Comment by Xaver-DaRed** [comment]
I /stared at you, yeah

**Comment by hooksta4** [comment]
Ah ok I see what you meant now. Removed the else statement as well from the quest lua because now it should default to default message.

Thanks!

**Comment by hooksta4** [comment]
@Xaver-DaRed can you do a one over on this one again? Thanks.

**Comment by Xaver-DaRed** [comment]
I would:
- Add a comment in default actions for the 2 guards stating that SOMETHING changes them, probably rank.
- Study if mob logic can be moved to the quest script. It doesnt convince me theres quest logic in there

To be clear, mob logic CAN be inside the quest script. Some missions do this

**Comment by hooksta4** [comment]
> I would:
> 
> * Add a comment in default actions for the 2 guards stating that SOMETHING changes them, probably rank.
> * Study if mob logic can be moved to the quest script. It doesnt convince me theres quest logic in there
> 
> To be clear, mob logic CAN be inside the quest script. Some missions do this

Ok I will tinker with it and see what I can come up with. It DOES make sense to put the old sabertooth stuff in the script file however, it might look different. Thanks

File: scripts/zones/Windurst_Woods/npcs/Perih_Vashai.lua [hunk 3]
```diff
@@ -4,7 +4,7 @@
 -- Starts and Finishes Quest: The Fanged One, From Saplings Grow
 -- !pos 117 -3 92 241
 -----------------------------------
-local ID = zones[xi.zone.WINDURST_WOODS]
+--local ID = zones[xi.zone.WINDURST_WOODS]
```
Comment (Xaver-DaRed) [inline scripts/zones/Windurst_Woods/npcs/Perih_Vashai.lua]: If this line is not needed anymore, just remove it.

**Comment by Xaver-DaRed** [comment]
Looks, good to go. Just adress that one comment I left.

**Review by Xaver-DaRed — APPROVED** [review]

**Comment by Xaver-DaRed** [comment]
This will need a rebase. Sorry mate

**Comment by hookstar** [comment]
> This will need a rebase. Sorry mate

what got changed?

**Comment by hooksta4** [comment]
> This will need a rebase. Sorry mate

Oh the proposal quest.status stuff....

**Comment by hooksta4** [comment]
Updated.

**Pull request closed without merging**